src/shared/models/NotificationTypes.ts

```diff
@@ -0,0 +1,25 @@
+import { CreatePayload } from '../../client/internal';
+
+export enum NotificationType {
+    DEFAULT = 'DEFAULT',
+    BUTTON = 'BUTTON',
+    INLINE = 'INLINE',
+    INLINEBUTTON = 'INLINEBUTTON'
+}
+
+export function resolveType(payload: CreatePayload): NotificationType {
```

Comment:
Why the near duplication between this and resolveTypes in Notification.ts?

Reply:
I'm part-way through restructuring most of this stuff. It will be removed soon, so shouldn't really matter.